File: go/space-age/space_age.go
// Package space provides a method for calculating the age on the planet
package space

// Planet represents a planet as a string
type Planet string

// Age calculates how old someone would be on a given planet
func Age(s float64, p Planet) float64 {
	const earthPeriod = 31557600.0

	planetsPeriod := map[Planet]float64{
		"Mercury": 0.2408467 * earthPeriod,
		"Venus":   0.61519726 * earthPeriod,
		"Earth":   earthPeriod,
		"Mars":    1.8808158 * earthPeriod,
		"Jupiter": 11.862615 * earthPeriod,
		"Saturn":  29.447498 * earthPeriod,
		"Uranus":  84.016846 * earthPeriod,
		"Neptune": 164.79132 * earthPeriod,
	}

	return s / planetsPeriod[p]
}

File: typescript/space-age/space-age.ts
const Planet = {
  Earth: 1.0,
  Mercury: 0.2408467,
  Venus: 0.61519726,
  Mars: 1.8808158,
  Jupiter: 11.862615,
  Saturn: 29.447498,
  Uranus: 84.016846,
  Neptune: 164.79132,
} as const

const EarthPeriodInSeconds = 31_557_600

class SpaceAge {
  constructor(public readonly seconds: number) {}

  onEarth = (): number => this.convert(Planet.Earth)
  onMercury = (): number => this.convert(Planet.Mercury)
  onVenus = (): number => this.convert(Planet.Venus)
  onMars = (): number => this.convert(Planet.Mars)
  onJupiter = (): number => this.convert(Planet.Jupiter)
  onSaturn = (): number => this.convert(Planet.Saturn)
  onUranus = (): number => this.convert(Planet.Uranus)
  onNeptune = (): number => this.convert(Planet.Neptune)

  private convert(ratio: number): number {
    return Math.round((this.seconds / (EarthPeriodInSeconds * ratio)) * 100) / 100
  }
}

export default SpaceAge
